src/coreclr/src/jit/flowgraph.cpp

```diff
@@ -23799,8 +23823,12 @@ Statement* Compiler::fgInlinePrependStatements(InlineInfo* inlineInfo)
         {
             const InlArgInfo& argInfo        = inlArgInfo[argNum];
             const bool        argIsSingleDef = !argInfo.argHasLdargaOp && !argInfo.argHasStargOp;
-            GenTree* const    argNode        = inlArgInfo[argNum].argNode;
-            unsigned __int64  bbFlags        = inlArgInfo[argNum].bbFlags;
+            GenTree*          argNode        = inlArgInfo[argNum].argNode;
+            const bool        argHasPutArg   = argNode->OperIs(GT_PUTARG_TYPE);
+
+            unsigned __int64 bbFlags = 0;
+            argNode                  = argNode->gtSkipPutArgType();
+            argNode                  = argNode->gtRetExprVal(&bbFlags);
```

Comment:
> See note above; I don't believe you can set these to zero like this and pass jit stress.

we read the flags here exactly how we read them in `impInlineInitVars` before but maybe I am missing something, will check.

Reply:
I think your changes are ok. Sorry for the false alarm.

Looks like the flags are captured when the `gtRetExpr` is created, and so by holding onto that node instead of tunneling through it, you no longer need to pass the flags around.